pkg/propertyprovider/azure/provider_integration_test.go

```diff
@@ -132,167 +132,175 @@ var (
+						Status:  metav1.ConditionTrue,
+						Reason:  CostPropertiesCollectionDegradedReason,
+						Message: fmt.Sprintf(CostPropertiesCollectionDegradedMsgTemplate, costCollectionWarnings),
+					},
 				}
-
-				var wantConditions []metav1.Condition
-				switch {
-				case costCollectionErr != nil:
-					wantConditions = []metav1.Condition{
-						{
-							Type:    CostPropertiesCollectionSucceededCondType,
-							Status:  metav1.ConditionFalse,
-							Reason:  CostPropertiesCollectionFailedReason,
-							Message: fmt.Sprintf(CostPropertiesCollectionFailedMsgTemplate, costCollectionErr),
-						},
-					}
-				case len(costCollectionWarnings) > 0:
-					wantConditions = []metav1.Condition{
-						{
-							Type:    CostPropertiesCollectionSucceededCondType,
-							Status:  metav1.ConditionTrue,
-							Reason:  CostPropertiesCollectionDegradedReason,
-							Message: fmt.Sprintf(CostPropertiesCollectionDegradedMsgTemplate, costCollectionWarnings),
-						},
-					}
-				default:
-					wantConditions = []metav1.Condition{
-						{
-							Type:    CostPropertiesCollectionSucceededCondType,
-							Status:  metav1.ConditionTrue,
-							Reason:  CostPropertiesCollectionSucceededReason,
-							Message: CostPropertiesCollectionSucceededMsg,
-						},
-					}
+			default:
+				wantConditions = []metav1.Condition{
+					{
+						Type:    CostPropertiesCollectionSucceededCondType,
+						Status:  metav1.ConditionTrue,
+						Reason:  CostPropertiesCollectionSucceededReason,
+						Message: CostPropertiesCollectionSucceededMsg,
+					},
 				}
+			}
 
-				expectedRes := propertyprovider.PropertyCollectionResponse{
-					Properties: wantProperties,
-					Resources: clusterv1beta1.ResourceUsage{
-						Capacity: corev1.ResourceList{
-							corev1.ResourceCPU:    totalCPUCapacity,
-							corev1.ResourceMemory: totalMemoryCapacity,
-						},
-						Allocatable: corev1.ResourceList{
-							corev1.ResourceCPU:    allocatableCPUCapacity,
-							corev1.ResourceMemory: allocatableMemoryCapacity,
-						},
-						Available: corev1.ResourceList{
-							corev1.ResourceCPU:    availableCPUCapacity,
-							corev1.ResourceMemory: availableMemoryCapacity,
-						},
+			expectedRes := propertyprovider.PropertyCollectionResponse{
+				Properties: wantProperties,
+				Resources: clusterv1beta1.ResourceUsage{
+					Capacity: corev1.ResourceList{
+						corev1.ResourceCPU:    totalCPUCapacity,
+						corev1.ResourceMemory: totalMemoryCapacity,
 					},
-					Conditions: wantConditions,
-				}
+					Allocatable: corev1.ResourceList{
+						corev1.ResourceCPU:    allocatableCPUCapacity,
+						corev1.ResourceMemory: allocatableMemoryCapacity,
+					},
+				},
+				Conditions: wantConditions,
+			}
 
-				res := p.Collect(ctx)
-				if diff := cmp.Diff(res, expectedRes, ignoreObservationTimeFieldInPropertyValue); diff != "" {
-					return fmt.Errorf("property collection response (-got, +want):\n%s", diff)
+			if isAvailableResourcesEnabled {
```

Comment:
do we have the negative test?

Reply:
Hi Ryan! Yeah, for integration tests we do; it's in the `feature gates` node of the integration test (line 988). One tests the case where costs are disabled, and the other where available resources are disabled.

Reply:
E2E test cases need to be added separately though; will submit subsequent PRs.